src/Components/App/FlagCard/FlagCard.js

```diff
@@ -1,18 +1,68 @@
-import React from 'react';
-import './FlagCard.css';
-import { Form } from '../../Form/Form';
-import { useLocation } from 'react-router-dom';
+import React from "react";
+import "./FlagCard.css";
+import { useLocation } from "react-router-dom";
+import { useState } from "react";
 
-export const FlagCard = ({ singleFlag, setSingleFlag, deleteFlag }) => {
+export const FlagCard = ({ singleFlag, deleteFlag, addFlag, getFlag}) => {
   const { id, flagPng, altText, name } = singleFlag;
-  const [alert, setAlert] = React.useState('');
-  const location = useLocation()
-  
+  const [alert, setAlert] = useState("");   
+  const [userAnswer, setUserAnswer] = useState("");
+  const [answer, setAnswer ] = useState("")
+  const location = useLocation();
+
+  const checkAnswer = (event) => {
+    event.preventDefault();
+    const alertMessage =
+      userAnswer !== name ? "You are incorrect, try again" : "You are correct!";
+    setAlert(alertMessage);
+    setUserAnswer("");
+  };
+
+  const saveFlag = () => {
+    const newFlag = singleFlag;
+    addFlag(newFlag);
+    setAlert("");
+    setAnswer("")
+    getFlag();
+  };
+
+  const showAnswer = () => {
+    setAnswer(name)
+    setAlert("")
+  }
+
+  const showNew = () => {
+    getFlag()
+    setAnswer("")
+  }
   return (
     <div className="flag-card">
-      <img src={flagPng} alt={altText}  />
-      <Form name={name} setSingleFlag={setSingleFlag} alert={alert} setAlert={setAlert} />
-      {location.pathname.includes('saved') && <button onClick={()=> {deleteFlag(name)}}>delete</button>}
+      <img src={flagPng} alt={altText} />
+      <form onSubmit={checkAnswer}>
+        <input
+          name="country"
+          type="text"
+          placeholder="What country?"
+          value={userAnswer}
```

Comment:
i think it would be worthwhile to allow the user to type in either lowercase or uppercase, and it should still work as intended. I noticed that when i used all lowercase, my answers would always fail